### ISSUE TITLE
Remove public client implementations from SDK

### ISSUE THREAD

**Issue by OlegRa** [alpacahq/alpaca-trade-api-csharp#327]
Right now SDK contains the only single pubic implementation of each `I...Client` interface. The recommended way for creating instances of these classes is fabric extension methods of the `IEnvorinment` interface. It shouldn't be a big problem for most users if we will make all these implementations internal.

**Comment by summerlinryan** [comment]
Hey Oleg, 

I'm starting to use this library and am really liking it so far. One thing I am trying to figure out is how to log all requests/responses from the underlying HttpClient used by the clients (e.g. AlpacaTradingClient). The solution I came up with was to create my own LoggingHandler that extends DelegatingHandler, then create a HttpClient that uses that LoggingHandler, then set HttpClient property of the AlpacaTradingClientConfiguration which will make the trading client use the HttpClient I provided. This solution requires me to be able to create an instance of AlpacaTradingClient directly because currently all of the Environment extension methods that create an AlpacaTradingClient create their own instance of AlpacaTradingClientConfiguration that I can't control the HttpClient of. 

![Screenshot from 2021-06-03 22-05-45](https://user-images.githubusercontent.com/28543855/120735044-e251c100-c4b7-11eb-8c69-e28aee7dff78.png)

Another solution I considered was trying to extend the CustomHttpHandler that exists in the ThrottleParameters class which I thought would be a good solution because it allows taking advantage of the IAsyncPolicy that uses Polly for retries. This solution couldn't work though because CustomHttpHandler is private.

So to summarize, could you either leave the clients public or could you expand the Environment extensions so that a custom configuration (with a custom HttpClient) could be provided to the clients? Currently, I see no way to provide a custom configuration to the clients other than by referencing the clients directly as I've done. 

Thanks!

**Comment by OlegRa** [comment]
@summerlinryan Yes, it was my fault - I've just never thought about non-standard use cases :)

I've added four factory methods for all client configuration classes. So now you can create an instance of any client using the `GetClient` method of the configuration class. But I still think it's not a common use case and therefore I've placed these factory methods into the `Alpaca.Markets.Extensions` package. Just use the latest [5.0.3-rc2](https://www.nuget.org/packages/Alpaca.Markets.Extensions/5.0.3-rc2) version of this package.

In your case the `tradingClient` creation will look like (after adding the proper `using` directive):

```cs
var tradingClient = tradingConfiguration.GetClient();
```
